src/asmcnc/comms/serial_connection.py

```diff
@@ -845,6 +855,49 @@ def process_grbl_push(self, message):
                     self.feed_rate = feed_speed[0]
                     self.spindle_speed = feed_speed[1]
 
+                # TEMPERATURES
+                elif part.startswith('TC:'):
+                    temps = part[3:].split(',')
+
+                    try: 
+                        float(temps[0])
+                        float(temps[1])
+                    except:
+                        log("ERROR status parse: Temperature invalid: " + message)
+                        return
+
+                    self.pcb_temp = float(temps[0])
+                    self.motor_driver_temp = float(temps[1])
+
+                    try:
+                        float(temps[2])
+                        self.transistor_heatsink_temp = float(temps[2])
+
+                    except IndexError:
+                        pass
+
+                    except Exception as E:
+                        log("ERROR status parse: Temperature invalid: " + message)
+                        return
+
+                # VOLTAGES
+                elif part.startswith('V:'):
+                    voltages = part[2:].split(',')
+                    try: 
+                        float(voltages[0])
+                        float(voltages[1])
+                        float(voltages[2])
+                        float(voltages[3])
+
+                    except:
+                        log("ERROR status parse: Voltage invalid: " + message)
+                        return
+
+                    self.microcontroller_mV = float(voltages[0])
+                    self.LED_mV = float(voltages[1])
+                    self.PSU_mV = float(voltages[2])
+                    self.spindle_speed_monitor_mV = float(voltages[3])
+
                 else:
                     continue
```

Comment:
is this redundant?

Reply:
not sure what you mean? which bit?

Reply:
the else: which only contains a continue - is it necessary? code would continue regardless

Reply:
ooh, no I don't think it is - just was always in there so I left it 😂